src/lib/execPromise.ts

```diff
@@ -126,32 +136,31 @@ export const execPromise = ({
       }
 
       if (stderr.includes("linker `cc` not found")) {
-        rej(
-          new CLIError(
-            `\n${color.yellow(MARINE_CARGO_DEPENDENCY)} requires ${color.yellow(
-              "build-essential"
-            )} to be installed. Please install it and try again.\nOn debian-based systems (e.g. Ubuntu) you can install it using this command:\n\n${color.yellow(
-              "sudo apt install build-essential"
-            )}\n`
-          )
-        );
-
+        const expectedErrorMessage = `\n${color.yellow(
+          MARINE_CARGO_DEPENDENCY
+        )} requires ${color.yellow(
+          "build-essential"
+        )} to be installed. Please install it and try again.\nOn debian-based systems (e.g. Ubuntu) you can install it using this command:\n\n${color.yellow(
+          "sudo apt install build-essential"
+        )}\n`;
+
+        res(new CLIError(expectedErrorMessage));
         return;
       }
 
       if (code !== 0) {
-        rej(
-          new Error(
-            `${printOutput ? "" : stderr}\nprocess exited${
-              code === null ? "" : ` with code ${code}`
-            }\n${debugInfo}`
-          )
-        );
-
-        return;
+        const commandFailedMessage = getCommandFailedMessage(code);
+        const errorMessage = getErrorMessage(printOutput, stderr);
+        res(new Error(`${commandFailedMessage}${errorMessage}`));
       }
 
       res(stdout);
     });
   });
+
+  if (result instanceof CLIError || result instanceof Error) {
```

Comment:
Aren't all `CLIError` are instances of `Error`?

Reply:
Yeah. It is. Wanted to be overprotective in case CLIError implementation changes for some reason but seems like typescript would catch that issue for me. So it's good to remove additional check